File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int[] before = new int[arr.length];
		int ans = 0;
		int cnt = 0;

		while (cnt != before.length) {
			cnt = 0;
			before = arr.clone();

			for (int i = 0; i < arr.length; i++) {
				if (arr[i] >= 50 && arr[i] % 2 == 0) arr[i] /= 2;
				else if (arr[i] < 50 && arr[i] % 2 != 0) arr[i] = arr[i] * 2 + 1;

				if (arr[i] == before[i]) cnt++;
			}

			ans++;
		}

		return ans - 1;
    }
}

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.kt
class Solution {
    fun solution(arr: IntArray): Int {
        var answer: Int = 0
        while (func(arr)) answer++
        return answer
    }

    private fun func(arr: IntArray): Boolean {
        var flag = false

        arr.indices.forEach {
            if (arr[it] >= 50 && arr[it] % 2 == 0) {
                arr[it] /= 2
                flag = true
            } else if (arr[it] < 50 && arr[it] % 2 == 1) {
                arr[it] = arr[it] * 2 + 1
                flag = true
            }
        }

        return flag
    }
}